### DISCUSSION TITLE
Collision shapes for gltf model for physics engine

### DISCUSSION THREAD

**Discussion by azhong-git** [google/filament#7214] [General]
Hi Filament community,

I am curious if there are some recommended ways or future plans for getting collision shapes (like convex hull or triangle mesh) from gltf model for physics simulation. As of today, the `AssetLoader` only computes AABB bounds for nodes.

Ideally in C++ (I know a few javascript libraries can do this)

**Comment by romainguy** [comment]
There are no plans for this as Filament is purely a rendering engine. However you can use a library like [Bullet](https://github.com/bulletphysics/bullet3) to drive physics and compute the convex hull of any mesh. It's efficient and easy to use, highly recommended.

**Reply by azhong-git** [reply]
Thank you for suggesting Bullet. Can I get mesh from Filament to feed Bullet? (It seems someone else also asked the same question. Therefore I created a separate Q&A for this https://github.com/google/filament/discussions/7219, hope you don't mind)

(I use `giltfio::AssetLoader` and `createInstancedAsset` to get `gltfio::FilamentAsset` but it seems to me I cannot get the raw mesh from this workflow).
